care/utils/assetintegration/onvif.py

```diff
@@ -0,0 +1,53 @@
+import enum
+import json
+from urllib import request
+
+from care.utils.assetintegration.base import BaseAssetIntegration
+
+
+class OnvifAsset(BaseAssetIntegration):
+    _name = "onvif"
+
+    class OnvifActions(enum.Enum):
+        MOVE_ABSOLUTE = "move_absolute"
+        GOTO_PRESET = "goto_preset"
+
+    def __init__(self, meta):
+        try:
+            self.meta = json.loads(meta)
+            self.meta = meta
+            self.name = meta["camera_type"]
+            self.host = meta["camera_address"]
+            self.port = meta["camera_port"] or 80
+            self.username = meta["camera_access_key"].split(":")[0]
+            self.password = meta["access_credentials"].split(":")[1]
+            self.middleware_hostname = meta["middleware_hostname"]
+        except KeyError:
+            print("Error: Invalid Onvif Asset")
```

Comment:
A bit more information on why it's an invalid asset would be great, this should be raising an exception.

Reply:
Switched to "Error: Invalid Onvif Asset; Missing required fields"